fx2eeprom.c

```diff
@@ -123,10 +126,13 @@ int main(int argc, char *argv[]){
 	}
 	else {
 		ret=fread(buffer,1,length,stdin);
-		if (length!=ret){
-			fprintf(stderr,"Wrong size from stdin, expected %d read %d\n",length,ret);
-			perror("fread");
-			return -1;
+		if (ret < length){ /* too few bytes? */
```

Comment:
What if fread returns an error?. I would leave the check that we currently have. If the user provided a file of a smaller size it should be an error.

Reply:
This requires that you know the file size, in Makefiles it needs a previous step with `SIZE=$(wc -c ${INFILE})`.